Modules/rats.py

```diff
@@ -71,14 +74,30 @@ def __eq__(self, other: 'Rats')->bool:
 
         Returns:
             bool: equal if uuid, platform, and name match
+
+        Raises:
+            TypeError : invalid type for equality check.
         """
+        if other is None:
+            return False
+
+        if not isinstance(other, Rats):
+            raise TypeError
```

Comment:
Here's a thing I've recently come across: As part of the `__eq__` contract in Python, [it should return `NotImplemented`](https://docs.python.org/3/reference/datamodel.html#object.__eq__) if the class doesn't know how to compare itself to the other object. Python can then try using the `__eq__` method of the other object or return `False` if neither object knows how to compare itself to the other one.